components/org.wso2.carbon.identity.verification.onfido.api/org.wso2.carbon.identity.verification.onfido.api.v1/src/main/java/org/wso2/carbon/identity/verification/onfido/api/v1/core/OnfidoIdvService.java

```diff
@@ -0,0 +1,536 @@
+            throw new OnfidoClientException(ERROR_INVALID_OR_MISSING_DATA_COMPARISON.getCode(),
+                    ERROR_INVALID_OR_MISSING_DATA_COMPARISON.getMessage());
+        }
+
+        return (Map<String, Object>) dataComparisonResults;
+    }
+
+    /**
+     * Updates the identity verification claims based on the provided Onfido verification request.
+     *
+     * @param verifyRequest The Onfido verification request containing the workflow run details and attribute
+     *                      verification results.
+     * @param idvpId        The identity verification provider ID.
+     * @param tenantId      The tenant ID.
+     * @param idVProvider   The identity verification provider.
+     * @throws OnfidoClientException If the payload structure is invalid or required data is missing.
+     * @throws OnfidoServerException If a server-side error occurs while updating the claims.
+     */
+    private void updateIdVClaims(VerifyRequest verifyRequest, String idvpId, int tenantId, IdVProvider idVProvider)
+            throws OnfidoClientException, OnfidoServerException {
+
+        VerifyRequestPayload payload = verifyRequest.getPayload();
+        String workflowRunId = payload.getObject().getId();
+        String completedAt = payload.getObject().getCompletedAtIso8601();
+        OnfidoConstants.WorkflowRunStatus workflowRunStatus =
+                OnfidoConstants.WorkflowRunStatus.fromString(payload.getObject().getStatus());
+        Map<String, Object> attributeVerificationResults = extractDataComparisonResults(payload.getResource());
+
+        try {
+            // Retrieve the WSO2 identity verification claims associated with the given workflow run ID.
+            // Note: The workflow run ID is unique per user, so this should return claims for a single user.
+            IdVClaim[] idVClaims = OnfidoIdvServiceHolder.getIdentityVerificationManager()
+                    .getIdVClaimsByMetadata(ONFIDO_WORKFLOW_RUN_ID, workflowRunId, idvpId, tenantId);
+            if (idVClaims == null || idVClaims.length == 0) {
+                throw new OnfidoClientException(ERROR_RETRIEVING_CLAIMS_AGAINST_WORKFLOW_RUN_ID.getCode(),
+                        ERROR_RETRIEVING_CLAIMS_AGAINST_WORKFLOW_RUN_ID.getMessage());
+            }
+
+            // Get the mapping of WSO2 claim URIs to Onfido claim names
+            // (e.g., "http://wso2.org/claims/lastname" maps to "last_name").
+            Map<String, String> claimMappings = idVProvider.getClaimMappings();
+
+            for (IdVClaim idVClaim : idVClaims) {
+
+                // Get the Onfido claim name corresponding to the WSO2 claim URI.
+                String wso2ClaimUri = idVClaim.getClaimUri();
+                String onfidoClaimName = claimMappings.get(wso2ClaimUri);
+                if (onfidoClaimName == null) {
+                    log.error(String.format("No mapped Onfido claim name identified for the claim URI: %s",
```

Comment:
This is a client error right?

Reply:
Decided to log this instead of throwing a client error because this API response is intended for the Onfido side. Unless the admin checks it on the Onfido dashboard, this issue won't be noticeable. Logging it ensures that we have a record of the discrepancy without disrupting the flow. We used error logs instead of debug logs to emphasize that this needs immediate attention from the admin. Additionally, the absence of a claim mapping for one attribute should not disrupt the verification status update for the rest.